Update Amazon S3 connector docs with latest release (#4744)
wso2/docs-apim: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md

{!reference/connectors/importing-connector-to-integration-studio.md!}

## Creating the Integration Logic

1. Specify the API name as `S3ConnectorTestAPI` and API context as `/s3connector`.

2. First we will create the `/createbucket` resource. This API resource will retrieve the bucket name from the incoming HTTP PUT request and create a bucket in Amazon S3.

> Right click on the API Resource and go to **Properties** view. We use a URL template called `/createbucket` as we have three API resources inside a single API. The method will be PUT.

Reviewer: I don't think the numbering will work unless we tab this.

Author: It's a single step, I have mistakenly added a new line. Fixed with a27bfd5

    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-connector-create-bucket-api.jpg" title="Adding the createbucket resource" width="800" alt="Amazon S3 use case"/>

> 3. Next drag and drop the 'createBucket' operation of the S3 Connector to the Design View as shown below. Here we are going to receive the following inputs from the user,

Reviewer: ```suggestion
3. Next drag and drop the 'createBucket' operation of the S3 Connector to the Design View as shown below. Here, you will receive the following inputs from the user.
```

    - bucketName - Name of the bucket

    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-connector-drop-create-bucket.jpg" title="Adding the createbucket operation" width="800" alt="Amazon S3 use case"/>

4. Create a connection from the properties window by clicking on the '+' icon as shown below.

    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-create-bucket-connection-configs.jpg" title="Creating a new connection" width="800" alt="Amazon S3 use case"/>

    In the popup window, the following parameters must be provided.

    - Connection Name - Unique name to identify the connection by.

>     - Connection Type - Type of the connection which specifies the protocol to be used.

Reviewer: ```suggestion
    - Connection Type - Type of the connection that specifies the protocol to be used.
```

    - AWS Access Key ID - Access key associated with your Amazon user account.
    - AWS Secret Access Key - Secret Access key associated with your Amazon user account.

>     - Region - Region which is used to select a regional endpoint to make requests

Reviewer: ```suggestion
    - Region - Region that is used to select a regional endpoint to make requests
```

    !!! note

>         1. You can either defince the credentials or let the the AWS SDK to manage the credentials. The SDK will look for AWS credentials in system/user environment variables or use the IAM role for authentication if the application is running in an EC2 instance.

Reviewer: ```suggestion
        1. You can either define the credentials or allow the AWS SDK to manage the credentials. The SDK will look for AWS credentials in system/user environment variables or use the IAM role for authentication if the application is running in an EC2 instance.
```

### Creating a bucket in Amazon S3

> 1. Create a file called `data.json` with the following content. Note that the bucket region is `us-east-2`. If you need to create the bucket in a different region, modify the hard coded region of the API configuration accordingly.

Reviewer: The us-east-2 region config isn't there in the json sample below. Can we modify this?

Author: As mentioned, the us-east-2 is hardcoded in the API, therefore we don't need to send it in the JSON payload.
